### ISSUE TITLE
Link to json is about:blank in swagger-ui

### ISSUE THREAD

**Issue by Wicpar** [nielsfalk/ktor-swagger#27]
Hello,

I've been fiddling with the project to see if it is a viable tool for my project, and I've come across a somewhat odd issue: The link to the openapi.json is not defined in the swagger-ui page.

How would i go about doing that?

**Comment by JLLeitschuh** [comment]
Can you elaborate?

**Comment by Wicpar** [comment]
Look on your heroku example when you click the link to the json, it is about:blank and not a real link.

**Comment by JLLeitschuh** [comment]
I'm not certain what causes that. If you know, PRs are welcome.

**Comment by Wicpar** [comment]
i fixed it by adding servers in the swagger file, maybe add a config entry for that?

**Comment by volkert-fastned** [comment]
@JLLeitschuh @Wicpar This is apparently an upstream issue in Swagger UI, having to do with relative URLs: https://github.com/swagger-api/swagger-ui/issues/4107

As a manual workaround, in the browser's URL bar you can prefix `swagger.json` or `openapi.json` (the part just behind `url=`) with `./` to force the link to the file to be recognized as a relative URL:

- /index.html?url=openapi.json -> /index.html?url=./openapi.json
- /index.html?url=swagger.json -> /index.html?url=./swagger.json

In the ktor-swagger source code, this workaround could be applied programmatically in `SwaggerSupport.kt`, specifically in the following extension function:

```
private suspend fun PipelineContext<Unit, ApplicationCall>.redirect(path: String, defaultJsonFile: String) {
    call.respondRedirect("/$path/index.html?url=$defaultJsonFile")
}
```
Direct link to the line to be fixed: https://github.com/nielsfalk/ktor-swagger/blob/fc247bc3e1b4bf73f0d720deb361d23cb7b925e1/ktor-swagger/src/main/kotlin/de/nielsfalk/ktor/swagger/SwaggerSupport.kt#L91

Basically, one would have to add the "./" prefix behind `url=` in the string argument within the above extension function as well. However, although it wouldn't hurt to add this, perhaps it's preferable to wait for an upstream version of Swagger UI that has this fixed instead. What do you think, @JLLeitschuh? I can make a PR for this if you want.

**Comment by JLLeitschuh** [comment]
I'd be happy to accept a PR for this fix.

**Comment by volkert-fastned** [comment]
OK, I'll make one. Could you please also directly publish a new release to Maven Central and/or JCenter after merging it? That would be much appreciated.

**Comment by volkert-fastned** [comment]
It would be a minor fix, so a minor version bump to 0.4.1 would suffice, I think.

**Comment by volkert-fastned** [comment]
@JLLeitschuh @Wicpar PR created: https://github.com/nielsfalk/ktor-swagger/pull/31

**Comment by volkert-fastned** [comment]
Workaround has been merged to master. I guess this issue #27 can be closed as soon as a release has been published containing this fix.

**Comment by volkert-fastned** [comment]
@Wicpar You can now close this issue, since @JLLeitschuh has just published version 0.5.0 of ktor-swagger to JCenter, which contains this fix/workaround. Thanks.

- https://github.com/nielsfalk/ktor-swagger/pull/31#issuecomment-477288833
- https://bintray.com/ktor-swagger/maven-artifacts/ktor-swagger

**Comment by Wicpar** [comment]
Thank you
